File: algoritms/dijkstraAlgorithm.php
<?php

$graph1 = array(
    'start' => array(),
    'start' => array('a' => 6),
    'start' => array('b' => 2), 
    'a' => array(),
    'a' => array('fin' => 1), 
    'b' => array(),
    'b' => array('a' => 3),
    'b' => array('fin' => 5),  
    'fin' => array(), 
);

$graph = array(
    'start' => array(
        array('a' => 6),
        array('b' => 2),
    ),
    'a' => array(
       array('fin' => 1),
    ),
    'b' => array(
       array('a' => 3),
       array('fin' => 5),
    ),
    'fin' => array(
        array('fin' => 0),
    ),
);

$infinity = INF;

$costs = array(
    'a' => 6,
    'b' => 2,
    'fin' => $infinity,
);

$parents = array(
    'a' => 'start',
    'b' => 'start',
    'fin' => NULL,
);

$processed = array();

function find_lowest_cost_node($costs, $processed) {
    $lowest_cost = INF;
    $lowest_cost_node = NULL;
    
    foreach ($costs as $key => $val) {
        if($val < $lowest_cost && !isset($processed[$key])) {
            $lowest_cost = $val;
            $lowest_cost_node = $key;
        }  
    }
    return $lowest_cost_node;
}

$node = find_lowest_cost_node($costs, $processed);
while ($node <> NULL) {
    $cost = $costs[$node];
    $neighbors = $graph[$node];
    foreach ($neighbors as $keys => $values) {
        foreach ($values as $key => $val) {
            $new_cost = $cost + $val;
            if($costs[$key] > $new_cost) {
                $costs[$key] = $new_cost;
                $parents[$key] = $node;
            }
        }
    }
    $processed[$node] = TRUE;
    $node = find_lowest_cost_node($costs, $processed);
 
}

print_r($costs);

File: algoritms/dijkstraAlgorithm.java
package algoritms;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class dijkstraAlgorithm {
	
	private static Map<String, Map<String, Double>> graph = new HashMap<>();
	private static List<String> processed = new ArrayList<>();
	
	private static String findlowestCostNode(Map<String, Double> costs) {
		Double lowestCost = Double.POSITIVE_INFINITY;
		String lowestCostNode = null;
		
		for(Map.Entry<String, Double> node: costs.entrySet()) {
			Double cost = node.getValue();
			if (cost < lowestCost && !processed.contains(node.getKey())) {
				lowestCost = cost;
				lowestCostNode = node.getKey();
			}
		}
		
		return lowestCostNode;
		
	}

	public static void main(String[] args) {
		
		graph.put("start", new HashMap<>());
		graph.get("start").put("a", 6.0);
		graph.get("start").put("b", 2.0);
		
		graph.put("a", new HashMap<>());
		graph.get("a").put("fin", 1.0);
		
		graph.put("b", new HashMap<>());
		graph.get("b").put("a", 3.0);
		graph.get("b").put("b", 5.0);
		
		graph.put("fin", new HashMap<>());
		
		Map<String, Double> costs = new HashMap<>();
		costs.put("a", 6.0);
		costs.put("b", 2.0);
		costs.put("fin", Double.POSITIVE_INFINITY);
		
		Map<String, String> parents = new HashMap<>();
		parents.put("a", "start");
		parents.put("b", "start");
		parents.put("fin", null);
		
		String node = findlowestCostNode(costs);
		while (node != null) {
			Double cost = costs.get(node);
			Map<String, Double> neighbors = graph.get(node);
			for (String n : neighbors.keySet()) {
				double newCost = cost + neighbors.get(n);
				if (costs.get(n) > newCost) {
					costs.put(n, newCost);
					parents.put(node,  node);
				}
			}
			processed.add(node);
			node = findlowestCostNode(costs);
		}
		
		System.out.println(costs);
		
	}

}
